OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -188,6 +187,25 @@ public virtual bool IsCloseEnoughToBase(World world, Player p, string buildingNa
 			var bi = world.WorldActor.Trait<BuildingInfluence>();
 			var allyBuildEnabled = world.WorldActor.Trait<MapBuildRadius>().AllyBuildRadiusEnabled;
 
+			Func<Actor, bool> hasAreaTypes = a =>
```

Comment:
Please instead extract a function
```csharp
bool ActorGrantsValidArea(Actor a)
{
	return AreaTypes.Overlaps(a.TraitsImplementing<GivesBuildableArea>()
		.SelectMany(gba => gba.Types));
}
```

and give `BuildingInfo` and `GivesBuildableAreaInfo` a default value of `"Buildable"` in their type lists.  The extra complexity of supporting an empty set granting all types isn't worthwhile.